doc/spec.md

```diff
@@ -3755,6 +3768,68 @@ y = set([2, 3])
 x.union(y)                              # set([1, 2, 3])
 ```
 
+<a id='set·add'></a>
+### set·add
+
+`S.add(x)` adds the value `x` to the set `S`. This has no effect if `x` is already in `S`.
+
+```python
+x = set([1, 2])
+x.add(3)                             # None (x == set([1, 2, 3]))
+x.add(3)                             # None (x == set([1, 2, 3]))
+```
+
+<a id='set·pop'></a>
+### set·pop
+
+`S.pop()` removes an arbitrary item from the set and returns it.
+
+`pop` fails if the set is empty.
+
+```python
+x = set([1, 2, 3])
+x.pop()                              # 1 (x == set([2, 3]))
+```
+
+<a id='set·remove'></a>
+### set·remove
+
+`S.remove(x)` removes `x` from the set and returns None.
+
+`remove` fails if the set does not contain `x` or is frozen.
+
+```python
+x = set([1, 2, 3])
+x.remove(2)                             # None (x == set([1, 3]))
+x.remove(2)                             # error: element not found
+```
+
+<a id='set·discard'></a>
+### set·discard
+
+`S.discard(x)` removes `x` from the set and returns None.
```

Comment:
Given my mutation comment above, it might be easier to express by cases. Also, when the return value is trivial (None) it can be relegated to a secondary paragraph. Something like:

> If x is an element of set S, S.discard(x) removes x from the set, or fails if the set is frozen. If x is not an element of the set, discard has no effect.

> It returns None.